ui/src/app/base/types.ts

```diff
@@ -136,6 +138,19 @@ export type Model = {
   id: number;
 };
 
+/**
+ * A named foreign model reference, e.g. machine.domain
+ */
+export type ModelRef = {
+  name: string;
+} & Model;
```

Comment:
I wonder if there's a convention for type unions, i.e. should it be `Model & { extra: "stuff" }` or `{ extra: "stuff" } & Model`?

Reply:
I think I prefer the former actually, for long type definitions, it makes it clearer having the existing type in the intersection first.